lib/request.js

```diff
@@ -1,41 +1,10 @@
 import core from "@actions/core";
 import { request } from "@octokit/request";
-import { ProxyAgent, fetch as undiciFetch } from "undici";
```

Comment:
I wonder if we need to explicitly use `EnvHttpProxyAgent`?
https://github.com/nodejs/undici/blob/7f635e51f6170f4b59abedc7cb45e6bcda7f056d/docs/docs/api/EnvHttpProxyAgent.md#L4

See this comment: https://github.com/nodejs/undici/issues/1650#issuecomment-2323919279

I just wanted to leave that not for when we have another look at it, I don't have time to dig into it now

Reply:
I tried that in https://github.com/actions/create-github-app-token/pull/143/commits/5cb4b2f8ba13156842321ca6bfda19c66ea0fc58, but several tests started failing as a result. After reading through https://github.com/nodejs/undici/pull/2994, I got the impression it was unnecessary to do this.

> Added `EnvHttpProxyAgent` and made it the default global dispatcher.

> Automatically detect/support HTTP_PROXY, HTTPS_PROXY, & NO_PROXY

Reply:
Super random drive-by comment. I got here from that issue. 😅 

According to this commit, they ended up not using this as the default agent:
https://github.com/nodejs/undici/pull/2994/commits/f31d3f9ca9266a836fa3267b140b00f5ee27463d

Reasoning:
https://github.com/nodejs/undici/pull/2994#discussion_r1540707186

So it appears the PR description is wrong. Looks like they'll consider making it the default in a major release.

For now, I was able to get HTTPS_PROXY env variable to work via:
```js
import { EnvHttpProxyAgent, setGlobalDispatcher } from "undici";

setGlobalDispatcher(new EnvHttpProxyAgent());
```

Reply:
Thanks, @bruce-c-liu! I found some additional context here:

https://github.com/nodejs/TSC/blob/41602b95211f30107696a007e5da45b9e933536c/meetings/2024-09-04.md?plain=1#L88-L115